railties/lib/rails/commands/server/server_command.rb

```diff
@@ -238,12 +284,35 @@ def pid
         end
 
         def self.banner(*)
-          "rails server [puma, thin etc] [options]"
+          "rails server [thin/puma/webrick] [options]"
         end
 
         def prepare_restart
           FileUtils.rm_f(options[:pid]) if options[:restart]
         end
+
+        def capture_missing_handler(server)
+          server.handler
+        rescue Rails::Server::MissingError
+          yield
+        end
+
+        def notify_on_exit
+          yield
+        ensure
+          puts "Exiting" unless options[:daemon]
+        end
+
+        def print_boot_information(handler, ssl)
+          use_puma = handler.to_s == "Rack::Handler::Puma"
+          url = "on #{ssl ? 'https' : 'http'}://#{host}:#{port}" unless use_puma
```

Comment:
Let's let the server build this URL in a nodoc'ed `served_url`, so we can do `server.served_url` below.

Then we can keep `use_puma?`